src/provider/index.ts

```diff
@@ -12,4 +13,19 @@ export async function getProviderOrThrow(
   return provider;
 }
 
+export function getProviders(): WebbtcProvider[] {
+  if (!window.webbtc_providers) window.webbtc_providers = [];
+  return window.webbtc_providers;
+}
+
+export function getProviderById(providerId: string) {
+  if (Array.isArray(window.webbtc_providers)) {
+    const provider = window.webbtc_providers.find((provider) => provider.id === providerId);
+    return provider?.id?.split('.').reduce((acc: any, part) => acc?.[part], window);
+  } else {
+    console.error('window.webbtc_providers is not defined or not an array');
+    return null;
```

Comment:
It's not really an error if this isn't populated. We can just log a message. Also, would be good to return undefined because line 24 could potentially return undefined, so better to stay consistent.

```suggestion
    console.log('window.webbtc_providers is not defined or not an array');
    return undefined;
```